projects/ngrx.io/content/guide/store/feature-creators.md

```diff
@@ -73,6 +73,72 @@ export const selectBookListPageViewModel = createSelector(
 );
 </code-example>
 
+`createFeature` also can be used to provide extra selectors for the feature state with the `extraSelectors` option:
+
+<code-example header="books.feature.ts">
+import { createFeature, createReducer, on } from '@ngrx/store';
+import { Book } from './book.model';
+
+import * as BookListPageActions from './book-list-page.actions';
+import * as BooksApiActions from './books-api.actions';
+
+interface State {
+  books: Book[];
+  query: string;
+}
+
+const initialState: State = {
+  books: [],
+  query: '',
+};
+
+export const booksFeature = createFeature({
+  name: 'books',
+  reducer: createReducer(
+    initialState,
+    on(BookListPageActions.search, (state, action) => ({
+      ...state,
+      query: action.query,
+    })),
+  ),
+  extraSelectors: ({selectQuery, selectBooks}) => ({
+    selectFilteredBooks: createSelector(
+      selectQuery,
+      selectBooks,
+      (query, books) => books.filter(book => book.title.includes(query)),
+    ),
+  }),
+});
+</code-example>
+
+The `extraSelectors` option accepts a function that takes the generated selectors as input arguments and returns an object of all the extra selectors. Other selectors can be "merged" here as well, including the ones generated by an `EntityAdapter`:
```

Comment:
I think this should be better as a recipe (can be in a different PR) because we combine features from 2 different packages.

Reply:
Not removed, but moved to a recipe.

Reply:
I merged the examples, how do you find this version?

Reply:
Sorry, I wasn't clear enough.
This is not exactly how I imagined it.

I think we can add extraSelectors to the existing snippet at https://ngrx.io/guide/store/feature-creators#using-feature-creator

We can add a new page to the recipes to showcase how to use features with entity adapters, similar to https://ngrx.io/guide/store/recipes/injecting

Reply:
Oh, I see, let me work on it in the weekend

Reply:
@timdeschryver I changed it to only include a basic example. I will also write a recipe in a separate PR. Is it ok to open an issue for it? And is the current version ok?